federation/src/graphql/api/1_0/util/authenticateCommunity.ts

```diff
@@ -48,68 +51,91 @@ export async function startOpenConnectionCallback(
     const client = AuthenticationClientFactory.getInstance(fedComA)
 
     if (client instanceof V1_0_AuthenticationClient) {
-      const callbackArgs = new OpenConnectionCallbackArgs()
-      callbackArgs.oneTimeCode = oneTimeCode.toString()
-      // TODO encrypt callbackArgs.url with requestedCom.publicKey and sign it with homeCom.privateKey
-      callbackArgs.url = homeFedCom.endPoint.endsWith('/')
-        ? homeFedCom.endPoint + homeFedCom.apiVersion
-        : homeFedCom.endPoint + '/' + homeFedCom.apiVersion
+      const url = homeFedComB.endPoint.endsWith('/')
+        ? homeFedComB.endPoint + homeFedComB.apiVersion
+        : homeFedComB.endPoint + '/' + homeFedComB.apiVersion
+
+      const callbackArgs = new OpenConnectionCallbackJwtPayloadType(handshakeID, oneTimeCode, url)
       logger.debug(`Authentication: start openConnectionCallback with args:`, callbackArgs)
-      if (await client.openConnectionCallback(callbackArgs)) {
-        logger.debug('startOpenConnectionCallback() successful:', callbackArgs)
+      // encrypt callbackArgs with requestedCom.publicJwtKey and sign it with homeCom.privateJwtKey
+      const jwt = await encryptAndSign(callbackArgs, homeComB!.privateJwtKey!, comA.publicJwtKey!)
+      const args = new EncryptedTransferArgs()
+      args.publicKey = homeComB!.publicKey.toString('hex')
+      args.jwt = jwt
+      args.handshakeID = handshakeID
+      const result = await client.openConnectionCallback(args)
+      if (result) {
+        logger.debug('startOpenConnectionCallback() successful:', jwt)
       } else {
-        logger.error('startOpenConnectionCallback() failed:', callbackArgs)
+        logger.error('startOpenConnectionCallback() failed:', jwt)
       }
     }
   } catch (err) {
     logger.error('error in startOpenConnectionCallback:', err)
   }
+  logger.removeContext('handshakeID')
 }
 
 export async function startAuthentication(
+  handshakeID: string,
   oneTimeCode: string,
   fedComB: DbFedCommunity,
 ): Promise<void> {
+  logger.addContext('handshakeID', handshakeID)
```

Comment:
better to use a logger instance per function call, if you use context

Reply:
introduce method logger